### DISCUSSION TITLE
Officially publish a relint container image

### DISCUSSION THREAD

**Discussion by ghost** [codingjoe/relint#59] [Ideas]
Hi 👋 

Would be nice to have an official container image of relint published somewhere: it really helps designing reusable pipelines when building blocks are containerized (VS install deps python, node and everyone’s favorite language runtime ;) )

Many thanks

**Comment by codingjoe** [comment]
Hi @gui-don,

Thanks for reaching out. Though, I don't know if I agree with everything you said, especially about everyone's favourite language… that should be obviously RegEx, right? 😅

Anyhow, I think there is no issue building off any Python package. However, I am not sure, I fully get the use case. When you are talking about a container image, you mean docker? What's the benefit of a container of the package? What'd be your current application?

Cheers!
Joe

**Reply by ghost** [reply]
> Though, I don't know if I agree with everything you said, especially about everyone's favourite language… that should be obviously RegEx, right? 😅

Yes! Of course, of course :) They also advantageously replace sudoku when one’s bored! 

> What's the benefit of a container of the package? What'd be your current application?

It’s because of some kind of unspoken convention in the CI.CD world. Pipelines are run inside agents (gitlab, github, circle-ci, jenkins…) with the bare minimum installed. Two schools of thoughts with different advantages and drawbacks:

- Install all needed dependencies for a pipeline in a "custom agent image".
- Keep the agent as it is and use a container tool like docker or podman to run most pipeline actions within containers.

I'm gonna focus on the second method, because it allows to make pipelines very consistent, it increases portability and allow quick experimentation. All insanely good things in the world of testing & deploying. The only drawback would be performance (pulls of container images), but that is circumvented by proper caching and using slim container bases like debian-slim or alpine.

That’s where my suggestion come from.

I’m gonna need to test it so I’ll build my own image anyway, but it’s always nicer to have an official image from the actual code owner. Since I think relint has potential to be used in CI.CD contexts a lot, providing a container image makes a lot of sense in my opinion.

>  When you are talking about a container image, you mean docker?

Yes, docker or open container.

**Reply by ghost** [reply]
Here is an example project: https://gitlab.com/wild-beavers/docker/relint
With container being publicly available there: https://gitlab.com/wild-beavers/docker/relint/container_registry/5735352

Far from ideal since updates would be manual and this project, while 100% open source and auditable, is not as trustable as an "official" from the owner of the project.

**Reply by codingjoe** [reply]
I think I understand your point. However, I honestly must say, I have plenty of things to maintain, and I am not eager to maintain a docker image unless there is any sizable community interest.
Furthermore, it seems a bit uncommon for my native domain (Python). I wouldn't know of any large Python linter, that publishes an official docker image.

Since you are the first one to mention this, and you have a running container, I'd leave it at that until this sees some upvotes. I hope that OK with you.
